source-reminder/lambda_function.py

```diff
@@ -0,0 +1,146 @@
+records_data = sheet_instance.get_all_records()  # Get all records of the data of the instance.
+empty_line = {'Timestamp': '', 'Your Name': '', 'Headline ': '', 'Link': '',
+              'Date of Publication': '', 'Is this for the ProPublica/The Texas Tribune investigative unit?': '',
+              'Type': '', 'Topic': '', 'Byline': '', 'Name of Source': '', 'Type of Source (choose best option)': '',
+              'Is this story based on news made by this person?': '',
+              'Did you confirm with this source how they identify in terms of race and gender?': '',
+              'Any notes about how asking this source went?': '',
+              'Gender': '', 'Race/Ethnicity': '', 'Do you have another source? ': '', 'Email Address': ''}
+records_data.remove(empty_line)  # Edge case.
+start = date.today() - timedelta(weeks=3)
+end = date.today() - timedelta(weeks=1)
+
+# View the data for the relevant window of time.
+timeframe_records = sorted(
+    filter(lambda x: start <= datetime.strptime(x['Date of Publication'], '%m/%d/%Y').date() <= end, records_data),
+    key=lambda i: i['Date of Publication'])
+sheet_headlines = [k['Headline '] for k in timeframe_records]
+
+
+def request_data(limit, start_date, end_date):
+    query = {'limit': limit, 'start_date': start_date, 'end_date': end_date}
+    return requests.get(API_ENDPOINT, params=query)
+
+
+response = request_data(100, start, end)
+sorted_site = sorted(response.json()['results'], key=lambda i: i['pub_date'])
+site_headlines = sorted([x['headline'] for x in sorted_site])
+
+submitted = set(sheet_headlines) & set(site_headlines)
+missing = list(sorted(set([x.strip() for x in site_headlines]) - set([x.strip() for x in sheet_headlines])))
+missing_full = sorted([x for x in sorted_site if x['headline'] in missing], key=lambda i: i['headline'])
+
+
+def itemize(elements):
+    return " and ".join([", ".join(elements[:-1]), elements[-1]] if len(elements) > 2 else elements)
+
+
+def cell(headline, image_url):
+    return {
+        "type": "section",
+        "text": {
+            "type": "mrkdwn",
+            "text": headline
+        },
+        "accessory": {
+            "type": "image",
+            "image_url": image_url,
+            "alt_text": "test"
+        }
+    }
+
+
+divider = {
+    "type": "divider"
+}
+
+blocks = [
+    {
+        "type": "section",
+        "text": {
+            "type": "mrkdwn",
+            # TODO - Format dates properly.
+            "text": "We found *{} articles* from *{} to {}* that do not have source information documented.".format(len(missing_full), start, end)
+        },
+        "accessory": {
+            "type": "button",
+            "text": {
+                "type": "plain_text",
+                "text": "Submit Info",
+                "emoji": True
+            },
+            "value": "click_me_123",
+            "url": "https://docs.google.com/forms/d/e/1FAIpQLSd653OLiLaYL2xJANKMucEoBLgcyeYCTCVScDiUAv6_qX4iWA/viewform",
```

Comment:
+1 from Brandyn's slack note that this one makes sense as an env variable. That'd make it more clear to update too if we ever moved the form

Reply:
Seconding what Ashley said here and in Slack also about helping make the script more configurable/versatile/re-usable, which is awesome.  To help make it configurable, I like to think about which ENV Variables are the most likely to be changed / updated / configured and put them in a code block near the top so it's almost like documentation for the script user for what needs to be configured.